napari/_qt/qt_viewer.py

```diff
@@ -329,6 +326,9 @@ def __init__(self, viewer: ViewerModel, show_welcome_screen: bool = False):
         # bind shortcuts stored in settings last.
         self._bind_shortcuts()
 
+        for layer in self.viewer.layers:
```

Comment:
Fix looks good - I'd expecting the class to finish setting itself up before updating based on the layers.